engine/src/audio/plugin.cpp

```diff
@@ -0,0 +1,39 @@
+#include <cubos/core/al/audio_context.hpp>
+
+#include <cubos/engine/assets/plugin.hpp>
+#include <cubos/engine/audio/bridge.hpp>
+#include <cubos/engine/audio/plugin.hpp>
+#include <cubos/engine/settings/plugin.hpp>
+
+CUBOS_DEFINE_TAG(cubos::engine::audioInitTag);
+CUBOS_DEFINE_TAG(cubos::engine::audioBridgeTag);
+CUBOS_DEFINE_TAG(cubos::engine::audioTag);
+
+using cubos::core::al::AudioContext;
+
+void cubos::engine::audioPlugin(Cubos& cubos)
+{
+    cubos.depends(settingsPlugin);
+
+    cubos.component<AudioSource>();
+    cubos.component<AudioListener>();
+
+    cubos.startupTag(audioTag);
+    cubos.startupTag(audioInitTag).after(settingsTag);
+    cubos.startupTag(audioBridgeTag).after(audioInitTag).before(audioTag);
+
+    cubos.startupSystem("initalize Audio plugin").tagged(audioInitTag).call([](Assets& assets, Settings& settings) {
+        auto audioContext = AudioContext::create();
+
+        unsigned int listenerCount = static_cast<unsigned int>(
+            settings.getInteger("audio.listener.count", static_cast<int>(MA_ENGINE_MAX_LISTENERS)));
+
+        auto audioDevice = audioContext->createDevice("MiniAudioDevice", listenerCount);
```

Comment:
There should be no references to mini audio in the engine all, that's the point of the wrapper.
There should be a way of getting the default device name.
Maybe `audioContext->createDevice(listenerCount);` should use the default one?